executor/src/main/java/org/k8loud/executor/command/CommandExecutionServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package org.k8loud.executor.command;
+
+import lombok.extern.slf4j.Slf4j;
+import net.schmizz.sshj.SSHClient;
+import net.schmizz.sshj.common.IOUtils;
+import net.schmizz.sshj.connection.channel.direct.Session;
+import net.schmizz.sshj.connection.channel.direct.Session.Command;
+import net.schmizz.sshj.transport.verification.PromiscuousVerifier;
+import net.schmizz.sshj.userauth.keyprovider.KeyProvider;
+import org.apache.commons.codec.binary.Base64;
+import org.jetbrains.annotations.NotNull;
+import org.k8loud.executor.exception.CommandException;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.security.*;
+import java.security.interfaces.RSAPrivateCrtKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.RSAPublicKeySpec;
+
+import static org.k8loud.executor.exception.code.CommandExceptionCode.FAILED_TO_EXECUTE_COMMAND;
+import static org.k8loud.executor.exception.code.CommandExceptionCode.NON_ZERO_EXIT_CODE;
+
+@Slf4j
+@Service
+public class CommandExecutionServiceImpl implements CommandExecutionService {
+    private static final int COMMAND_EXIT_CHECK_SLEEP_MS = 100;
+
+    public CommandExecutionServiceImpl() {
+        // Fixes https://stackoverflow.com/questions/6559272/algid-parse-error-not-a-sequence
+        // But maybe it's not the best approach
+        java.security.Security.addProvider(
+                new org.bouncycastle.jce.provider.BouncyCastleProvider()
+        );
+    }
+
+    @Override
+    public String executeCommand(@NotNull String host, @NotNull Integer port, @NotNull String privateKey,
+                               @NotNull String user, @NotNull String command) throws CommandException {
+        try (SSHClient client = initClient(host, port, privateKey, user);
+             Session session = client.startSession()) {
```

Comment:
NIT: some lost space :P

Reply:
huh?

Reply:
Maybe my eyes are failing me but it looks like there is an addional space before that line

Reply:
Formatted by IntelliJ, makes sense imo

Reply:
ok, just checking